### PR TITLE
Fix in sendPacket to return error in the callback if qos > 0

### PR DESCRIPTION
**Pull request by v-divya110**
The sendPacket had cb as undefined in case of publish with qos > 0, reassigned the cb before returning in case of error.

### CODE REVIEW

File: test/abstract_store.js [hunk 1]
```diff
@@ -130,4 +130,18 @@ module.exports = function abstractStoreTest (build) {
...
+    const store = new Store()
+    store.put = function (packet, cb) {
+      process.nextTick(cb, new Error('oops there is an error'))
+    }
+
+    store.put({messageId: 43}, function (err) {
+      if (err) {
+        done()
+      }
+    })
```
Comment (mcollina) [inline test/abstract_store.js]: this is not testing what you think you are testing. As you have modified the client, you should test the client, not the store.

**Comment by mcollina** [comment]
CI didn't run for the last commit (not sure why). Can you also fix the conflicts with master?

File: test/abstract_client.js [hunk 2]
```diff
@@ -450,56 +449,70 @@ module.exports = function (server, config) {
...
+    })
 
-      client.on('connect', function () {
-        client.subscribe('test', function () {
-          subscribeCalled = true
-        })
-        client.publish('test', 'test', function () {
-          client.end(false, function () {
-            subscribeCalled.should.be.equal(true)
-            done()
```
Comment (mcollina) [inline test/abstract_client.js]: why all these changes in the test? Can you only add the new one?

Comment (v-divya110) [reply test/abstract_client.js]: I did not add any of those it only got indented. The test did not run otherwise

Comment (mcollina) [reply test/abstract_client.js]: Can you please restore them as they where on master?

Comment (v-divya110) [reply test/abstract_client.js]: umm okay.. so you want me to revert the commit ?

Comment (v-divya110) [reply test/abstract_client.js]: I don't get what I should be doing given that the current master test/abstract-client has many more changes in it. Should I simply revert the merge and commit it again with my addition in the test script?

Comment (mcollina) [reply test/abstract_client.js]: Yes, that would be the preferred path.

Comment (v-divya110) [reply test/abstract_client.js]: okay so I reverted the merge and checked in my code again.

**Comment by mcollina** [comment]
Can you please check in https://github.com/mqttjs/MQTT.js/pull/732/files?
I think there are still too many changes in there.
You might be better off in doing another branch and cherry pick your commits.

**Comment by v-divya110** [comment]
yes I think I'll probably close this branch and create a new branch followed by a new pull request. There seem to be too many changes in this and I don't see a way of restoring it. It'll be easier to create a new branch instead

**Pull request closed without merging**